vcd/datasource_vcd_lb_app_profile.go

```diff
@@ -0,0 +1,113 @@
+package vcd
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func datasourceVcdLBAppProfile() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceVcdLBAppProfileRead,
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the App Profile is located",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the App Profile is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the App Profile is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "App Profile name for lookup",
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "Protocol type used to send requests to the server. One of 'TCP', " +
+					"'UDP', 'HTTP' org 'HTTPS'",
+			},
+			"enable_ssl_passthrough": &schema.Schema{
+				Type:     schema.TypeBool,
+				Computed: true,
+				Description: "Enable SSL authentication to be passed through to the virtual " +
+					"server. Otherwise SSL authentication takes place at the destination address.",
+			},
+			"http_redirect_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Description: "The URL to which traffic that arrives at the destination address " +
+					"should be redirected. Only applies for types HTTP and HTTPS",
+			},
+			"persistence_mechanism": &schema.Schema{
```

Comment:
Is "mechanism" used by vCD/API? Maybe another way of calling it would be "persistence_mode"? Thought the more I read this PR the more "mechanism" seems appropriate, even if longer :)

Reply:
It's actually called "persistance" in the GUI. I'm just thinking if this gives enough context to the field name. I added machanism here as per the official vCD documentation.

Excerpt:

> Persistence - Specify a persistence **mechanism** for the profile. Persistence tracks and stores session data, such as the specific pool member that serviced a client request. This ensures that client requests are directed to the same pool member throughout the life of a session or during subsequent sessions. 


https://docs.vmware.com/en/vCloud-Director/9.0/com.vmware.vcloud.tenantportal.doc/GUID-A8E70787-093E-46D2-8045-381F12BCCDF8.html

Reply:
`persistence_mechanism` it is then, thanks for explaining!